### PR TITLE
Sort assetType alphabetically in related assets page

### PR DESCRIPTION
**Pull request by LBHCallumM**
## Summary of Changes

In the related asset page, assetTypes were not sorted. Sometimes the order changed between renders.

- Add sorting to assetType (alphabetical, dwelling last)

### CODE REVIEW

File: src/components/related-assets/utils.ts [hunk 1]
```diff
@@ -36,34 +36,35 @@ const removeHackneyHomesRelatedAsset = (relatedAssets: RelatedAsset[]) => {
...
-      if (uniqueAssetType === relatedAsset.type) sameTypeAssets.push(relatedAsset);
-    });
+  uniqueAssetTypes.forEach((assetType) => {
+    const sameTypeAssets = relatedAssets.filter(x => x.type === assetType)
 
     // Sort assets by addressLine1 (name)
-    sortAddressGeneric(sameTypeAssets, "name");
+    if (sameTypeAssets.length > 1) {
+      sortAddressGeneric(sameTypeAssets, "name");
+    }
```
Comment (LBHCallumM) [inline src/components/related-assets/utils.ts:64]: Save method call if only single asset

File: src/components/related-assets/utils.ts [hunk 2]
```diff
@@ -36,34 +36,35 @@ const removeHackneyHomesRelatedAsset = (relatedAssets: RelatedAsset[]) => {
...
 
   // Define how many asset types we're dealing with
-  const uniqueAssetTypes = new Set<string>([]);
-  relatedAssets.forEach((relatedAsset) => uniqueAssetTypes.add(relatedAsset.type));
-
-  // If present, move "Dwelling" to last (so dwellings appear below other asset types)
-  if (uniqueAssetTypes.has("Dwelling")) {
-    uniqueAssetTypes.delete("Dwelling");
-    uniqueAssetTypes.add("Dwelling");
-  }
```
Comment (LBHCallumM) [inline src/components/related-assets/utils.ts:52]: Moved to assetTypeSorter

File: src/components/related-assets/utils.ts [hunk 3]
```diff
@@ -36,34 +36,37 @@ const removeHackneyHomesRelatedAsset = (relatedAssets: RelatedAsset[]) => {
   return relatedAssets.filter((relatedAsset) => relatedAsset.id !== hackneyHomesGuid);
 };
 
-export const organiseRelatedAssetsByType = (relatedAssets: RelatedAsset[]): Object => {
+const assetTypeSorter = (a: string, b: string) => {
+  if (a === "Dwelling" && b !== "Dwelling") {
+    return 1; // "Dwelling" should be considered greater, so it comes last
```
Comment (ange47rm) [inline src/components/related-assets/utils.ts:41]: https://github.com/LBHackney-IT/mtfh-tl-e2e-tests/pull/317

Comment (ange47rm) [inline src/components/related-assets/utils.ts:41]: I don't think comments on lines 41 and 44 are meant to be the same, are they?

Comment (LBHCallumM) [reply src/components/related-assets/utils.ts:41]: I can remove

Comment (ange47rm) [reply src/components/related-assets/utils.ts:41]: No no I like the comments, I think they're helpful in this case, but they're the exact same. From the context it doesn't seem like they're meant to be. Shouldn't the comment on line 41 say "... so it comes first" ?

File: src/components/related-assets/utils.ts [hunk 4]
```diff
@@ -36,34 +36,37 @@ const removeHackneyHomesRelatedAsset = (relatedAssets: RelatedAsset[]) => {
...
 
   // For each AssetType, create a array of RelatedAsset[]
-  uniqueAssetTypes.forEach((uniqueAssetType) => {
-    const sameTypeAssets: RelatedAsset[] = [];
-
-    relatedAssets.forEach((relatedAsset) => {
-      if (uniqueAssetType === relatedAsset.type) sameTypeAssets.push(relatedAsset);
-    });
+  uniqueAssetTypes.forEach((assetType) => {
+    const sameTypeAssets = relatedAssets.filter((x) => x.type === assetType);
```
[the remark this answers is not in the document]
Comment (LBHCallumM) [reply src/components/related-assets/utils.ts:61]: True, but I feel like it's unnecessary for a 1 line function. The more descript variable name is more useful in the foreach.

**Review by ange47rm — APPROVED** [review]

File: src/components/related-assets/utils.ts (hunk 3 above)
Comment (LBHCallumM) [reply src/components/related-assets/utils.ts]: I replaced it with a single comment

**Pull request merged**